include/ck_tile/core/numeric/vector_type.hpp

```diff
@@ -58,15 +63,16 @@ using ext_vector_t = typename impl::ext_vector<T, N>::type;
 template <typename T>
 struct vector_traits
 {
-    using scalar_type                    = remove_cvref_t<T>;
+    using scalar_type =
+        std::conditional_t<std::is_same_v<remove_cvref_t<T>, pk_int4_t>, int8_t, remove_cvref_t<T>>;
     static constexpr index_t vector_size = 1;
 };
 
 // specialization for ext_vector_type()
 template <typename T, index_t N>
 struct vector_traits<T __attribute__((ext_vector_type(N)))>
 {
-    using scalar_type                    = T;
+    using scalar_type = std::conditional_t<std::is_same_v<T, pk_int4_t>, int8_t, T>;
```

Comment:
You could have specialization here for non POD types (like basic int, int16 etc) where you would do: 
```suggestion
    using scalar_type = typename T::type;
```

Reply:
Generally with conditional_t it is not possible because conditional_t evaluates the both of types (for true and false), If I use ::type then it will check for float::type/int::type etc. Which has not such member